### PR TITLE
bug(perf-issues): hook up N+1 Query detection rate again

### PR DESCRIPTION
**Pull request by mjq-sentry**
The `n_plus_one_db_detection_rate` project option was being used to disable N+1 detection on projects with fingerprint issues. At some point the N+1 Query detector stopped respecting that option. Hook it back up, and hook up the MN+1 Query detector to it as well (as they create the same type of issue). If we have a reason in the future to split that option into two, we might, but for now they seem to share fingerprinting issues (typically dynamic custom parent span descriptions).

### CODE REVIEW

**Review by DominikB2014 — APPROVED** [review]
Nice 👍

File: static/app/views/settings/projectPerformance/projectPerformance.tsx [hunk 1]
```diff
@@ -179,15 +179,6 @@ class ProjectPerformance extends AsyncView<Props, State> {
...
       },
-      {
-        name: 'n_plus_one_db_issue_rate',
-        type: 'range',
-        label: t('N+1 (DB) Issue Rate'),
-        min: 0.0,
-        max: 1.0,
-        step: 0.01,
-        defaultValue: 0,
-      },
```
Comment (DominikB2014) [inline static/app/views/settings/projectPerformance/projectPerformance.tsx:190]: @mjq-sentry Might want to seperate this out to another PR to decouple frontend and backend

Comment (mjq-sentry) [reply static/app/views/settings/projectPerformance/projectPerformance.tsx:190]: D'oh, of course. Good catch, will do 👍

Comment (mjq-sentry) [reply static/app/views/settings/projectPerformance/projectPerformance.tsx]: https://github.com/getsentry/sentry/pull/45383

**Pull request merged**